hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/ResizableBlockCache.java

```diff
@@ -31,4 +31,13 @@
    * @param size The max heap size.
    */
   void setMaxSize(long size);
+
+  /**
+   * Check if block type is meta or index block
+   * @param blockType block type of a given HFile block
+   * @return true if block type is non-data block
+   */
+  default boolean isMetaBlock(BlockType blockType) {
+    return blockType.getCategory() != BlockType.BlockCategory.DATA;
```

Comment:
On HFileBlock we will have a blockType mostly.  But I believe there may be some rare cases where this is not yet assigned..  So to be on safe side, can we pls add a null check also here. I know this is an existing method only.
blockType != null && ...